src/base/preferences.cpp

```diff
@@ -1977,5 +1979,9 @@ void Preferences::setAddNewTorrentDialogSavePathHistoryLength(const int value)
 void Preferences::apply()
 {
     if (SettingsStorage::instance()->save())
+    {
         emit changed();
+        if (m_credentialsChanged)
+            emit webCredentialsChanged();
```

Comment:
```suggestion
        if (m_credentialsChanged)
        {
            emit webCredentialsChanged();
            m_credentialsChanged = false;
        }
```

Reply:
Do you still need the changes in isessionmanager.h after applying the above suggestions?

Reply:
> Do you still need the changes in isessionmanager.h after applying the above suggestions?

What? The change in `isessionmanager.h` is for terminating sessions which bypassed authentication when the bypass is disabled or changed to possibly stricter than before.

Reply:
I think it would be surprising for users with authenticated sessions to find they are logged out after changing these options.

By the way, we may want to disallow unauthenticated sessions to change the username/password.